_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,865 @@
+Also in Kubernetes, the deployed manifests can be provided statically or as an artifact. In this example, manifests are provided as artifacts stored in GitHub.
+
+For Spinnaker to access and act on resources, like docker registries, cloud providers, and code repositories, different types of accounts must be enabled in Spinnaker configuration, along with the credentials. In addition, the configuration has the optional step of associating Spinnaker with a Kubernetes service account, allowing to restrict permissions over the cluster resources granted to Spinnaker.
+
+### Install Halyard
+
+As described on Spinnaker [docs](https://spinnaker.io/docs/setup/install/), the first step is to install [Halyard](https://spinnaker.io/docs/setup/install/halyard/). For a local install in MacOS:
+
+```shell
+curl -O https://raw.githubusercontent.com/spinnaker/halyard/master/install/macos/InstallHalyard.sh
+sudo bash InstallHalyard.sh
+```
+Verify the installation with:
+
+```shell
+hal -v
+```
+
+You must also install [`kubectl`](https://kubernetes.io/docs/tasks/tools/), the Kubernetes command line too, to run commands against the clusters.
+
+### Choose GKE for Spinnaker deployment
+
+The second step is to choose a cloud provider for the environment in which you will install Spinnaker. For Kubernetes, Spinnaker needs a `kubeconfig` file, to access and manage the cluster. For creating a `kubeconfig` for a GKE cluster, you must first create the cluster. GoogleCloud provides a [free tier](https://cloud.google.com/free) of their services that grants you $300 in free credits if you are a new user.
+
+After you signed up, install [`gcloud` CLI](https://cloud.google.com/sdk/docs/install). Following the process to the end, the last step is to run `gcloud init` and set up the authorization for the tool.
+
+Create the cluster for the Spinnaker deployment with the following line:
+```shell
+gcloud container clusters create spinnaker-cluster \
+--zone southamerica-east1-a \
+--machine-type n1-standard-4 \
+--enable-autorepair \
+--enable-autoupgrade
+```
+Then fetch the cluster credentials with:
+
+```shell
+gcloud container clusters get-credentials spinnaker-cluster --zone southamerica-east1-a
+```
+
+`get-credentials` will update a `kubeconfig` file with appropriate credentials and endpoint information to point kubectl at a specific cluster in Google Kubernetes Engine.
+
+Create a namespace for the Spinnaker cluster:
+
+```shell
+kubectl create namespace spinnaker
+```
+
+The Spinnaker documentation recommends creating a Kubernetes service account, using Kubernetes role definitions that restrict the permissions granted to the Spinnaker account. Create a `spinnaker` directory, and add the file `spinnaker-service-account.yml` with the following content:
```

Comment:
Can you please link to the Spinnaker documentation?

Reply:
Linked